### PR TITLE
x/gravity: add events for jailing and slashing

### PR DESCRIPTION
**Pull request by alexanderbez**
- Add missing events for when a validator is jailed and slashed
- Some minor cleanup and error handling

### CODE REVIEW

File: module/x/gravity/abci.go [hunk 1]
```diff
@@ -247,35 +262,60 @@ func outgoingTxSlashing(ctx sdk.Context, k keeper.Keeper) {
...
 							params.SlashFractionBatch,
 						)
 						k.StakingKeeper.Jail(ctx, valInfo.cons)
+
+						ctx.EventManager().EmitEvent(
+							sdk.NewEvent(
+								slashingtypes.EventTypeSlash,
+								sdk.NewAttribute(slashingtypes.AttributeKeyAddress, valInfo.cons.String()),
+								sdk.NewAttribute(slashingtypes.AttributeKeyJailed, valInfo.cons.String()),
+								sdk.NewAttribute(slashingtypes.AttributeKeyReason, types.AttributeMissingBridgeBatchSig),
```
Comment (jackzampolin) [inline module/x/gravity/abci.go:280]: This code needs to be cleaned up. I think we have 1. slashing for missed sigs (not just batch sigs, all the sigs) 2. slashing for oracle stuff and I think theres a third one too. We could eliminate a bunch of params.

**Comment by jackzampolin** [comment]
This PR looks great. Reminds me that there is a lot of cleanup in the params. May of those vars aren't used anymore and are redundent.

**Review by jackzampolin — APPROVED** [review]
This PR looks great! It reminds me that there is a lot of cleanup to do in the params for the

File: module/x/gravity/abci.go (hunk 1 above)
Comment (alexanderbez) [reply module/x/gravity/abci.go:280]: Not sure I totally follow @jackzampolin. There are only two places where I saw that we jail and slash, which I've addressed in this PR.

I merged this PR. If there is more follow up needed, can you create an issue and I'll look into it pls?

**Pull request merged**